deepparse/parser/address_parser.py

```diff
@@ -218,37 +417,72 @@ def _process_device(self, device: Union[int, str, torch.device]):
         else:
             raise ValueError("Device should be a string, an int or a torch device.")
 
-    def _model_factory(self, model: str) -> None:
+    def _set_data_transformer(self):
+        train_vectorizer = TrainVectorizer(self.vectorizer, self.tags_converter)  # vectorize to provide also the target
+        data_transform = DataTransform(train_vectorizer,
+                                       self.model_type)  # use for transforming the data prior to training
+        return data_transform
+
+    def _create_training_data_generator(self, dataset_container: DatasetContainerInterface, train_ratio: float,
+                                        batch_size: int, num_workers: int, seed: int) -> Tuple:
+        # pylint: disable=too-many-arguments
+        data_transform = self._set_data_transformer()
+
+        train_indices, valid_indices = indices_splitting(num_data=len(dataset_container),
+                                                         train_ratio=train_ratio,
+                                                         seed=seed)
+
+        train_dataset = Subset(dataset_container, train_indices)
+        train_generator = DataLoader(train_dataset,
+                                     collate_fn=data_transform.teacher_forcing_transform,
+                                     batch_size=batch_size,
+                                     num_workers=num_workers,
+                                     shuffle=True)
+
+        valid_dataset = Subset(dataset_container, valid_indices)
+        valid_generator = DataLoader(valid_dataset,
+                                     collate_fn=data_transform.output_transform,
+                                     batch_size=batch_size,
+                                     num_workers=num_workers)
+
+        return train_generator, valid_generator
+
+    def _model_factory(self, path_to_retrained_model: Union[str, None] = None) -> None:
```

Comment:
For a futur PR I think this should be a factory class outside of the addressParser because is starting to have a lot of responsibility

Reply:
I agree, this is the first step in that direction.